enterprise/frontend/src/embedding-sdk/components/public/InteractiveQuestion/context.tsx

```diff
@@ -1,83 +1,142 @@
+import type {
+  NavigateToNewCardParams,
+  SdkQuestionResult,
+} from "embedding-sdk/types/question";
+import { useDispatch } from "metabase/lib/redux";
 import type { Mode } from "metabase/visualizations/click-actions/Mode";
 import { getEmbeddingMode } from "metabase/visualizations/click-actions/lib/modes";
+import type Question from "metabase-lib/v1/Question";
 
-import {
-  type UseLoadQuestionParams,
-  useInteractiveQuestionData,
-  useLoadQuestion,
-} from "./hooks";
-
-type InteractiveQuestionContextType = {
+interface InteractiveQuestionContextType extends SdkQuestionResult {
   plugins: SdkPluginsConfig | null;
   mode: Mode | null | undefined;
   isQuestionLoading: boolean;
   resetQuestion: () => void;
   onReset?: () => void;
   onNavigateBack?: () => void;
-};
+  onQuestionChange: (question: Question) => void;
+  onNavigateToNewCard: (params: NavigateToNewCardParams) => void;
+}
 
 /**
  * Note: This context should only be used as a wrapper for the InteractiveQuestionResult
  * component. The idea behind this context is to allow the InteractiveQuestionResult component
  * to use components within the ./components folder, which use the context for display
- * and functions. Any data that can be referenced from the store should be placed in
- * the useInteractiveQuestionData hook.
+ * and functions.
  * */
 export const InteractiveQuestionContext = createContext<
   InteractiveQuestionContextType | undefined
 >(undefined);
 
 type InteractiveQuestionProviderProps = PropsWithChildren<
-  UseLoadQuestionParams & {
+  {
     componentPlugins?: SdkPluginsConfig;
     onReset?: () => void;
     onNavigateBack?: () => void;
-  }
+  } & LoadSdkQuestionParams
 >;
 
 export const InteractiveQuestionProvider = ({
-  children,
   location,
   params,
   componentPlugins,
   onReset,
   onNavigateBack,
+  children,
 }: InteractiveQuestionProviderProps) => {
-  const { isQuestionLoading, resetQuestion } = useLoadQuestion({
-    location: location,
-    params: params,
-  });
+  const dispatch = useDispatch();
+
+  const [result, setQuestionResult] = useState<SdkQuestionResult>({});
+  const [isQuestionLoading, setIsQuestionLoading] = useState(true);
 
-  const { question } = useInteractiveQuestionData();
+  const { question, queryResults } = result;
 
   const globalPlugins = useSdkSelector(getPlugins);
   const plugins = componentPlugins || globalPlugins;
-  const mode = useMemo(
-    () => question && getEmbeddingMode(question, plugins || undefined),
-    [plugins, question],
+
+  const mode = question && getEmbeddingMode(question, plugins ?? undefined);
+
+  const storeQuestionResult = async (
+    getQuestionResult: () => Promise<SdkQuestionResult | null>,
+  ) => {
+    setIsQuestionLoading(true);
+
+    try {
+      const result = await getQuestionResult();
+
+      if (result) {
+        setQuestionResult(result);
+      }
+    } catch (e) {
+      console.error(`Failed to update question result`, e);
```

Comment:
We seem to already handle error states for query errors as part of query results.

We could auto return a `<SdkError />` here in case something unexpected happens (e.g. network errors) - at the moment it ignores unexpected errors.